build/commands/lib/test.js

```diff
@@ -63,11 +64,18 @@ const test = (passthroughArgs, suite, buildConfig = config.defaultBuildConfig, o
 
   braveArgs = braveArgs.concat(passthroughArgs)
 
+  let buildTarget = suite
+  if (suite === 'brave_unit_tests' || suite === 'brave_browser_tests')
+    buildTarget = "brave/test:" + suite
+
+  config.buildTarget = buildTarget
+
   // Build the tests
-  if (suite === 'brave_unit_tests' || suite === 'brave_browser_tests') {
-    util.run('ninja', ['-C', config.outputDir, "brave/test:" + suite], config.defaultOptions)
+  if (config.targetOS === 'android' && suite === 'brave_browser_tests') {
+    // Only android browser tests use this standard build function, to minimize impact on other tests
+    build(buildConfig, options)
```

Comment:
I don't understand what's going on here, all this does is run a bunch of extra stuff that we don't need for test builds

Reply:
It is because currently the command doesn't init and build the test automatically (it doesn't run `gn` and all the copying of files, so you need to run `npm run build` first). Also it makes it possible to build the tests with `goma`.